domain/src/main/java/org/oppia/android/domain/hintsandsolution/HintHandlerImpl.kt

```diff
@@ -86,31 +86,23 @@ class HintHandlerImpl private constructor(
   ) {
     handlerLock.withLock {
       when (helpIndex.indexTypeCase) {
-        NEXT_AVAILABLE_HINT_INDEX -> {
-          lastRevealedHintIndex = helpIndex.nextAvailableHintIndex - 1
-          latestAvailableHintIndex = helpIndex.nextAvailableHintIndex
+        NEXT_AVAILABLE_HINT_INDEX, LATEST_REVEALED_HINT_INDEX -> {
+          if (helpIndex.indexTypeCase == NEXT_AVAILABLE_HINT_INDEX) {
+            lastRevealedHintIndex = helpIndex.nextAvailableHintIndex - 1
+            latestAvailableHintIndex = helpIndex.nextAvailableHintIndex
+          } else {
+            lastRevealedHintIndex = helpIndex.latestRevealedHintIndex
+            latestAvailableHintIndex = helpIndex.latestRevealedHintIndex
+          }
```

Comment:
Ah, I misread the code before. Honestly, it's probably more readable the other way since we can't simplify it like the solution cases.

Reply:
done